lib/text/ui_text_displayer.js

```diff
@@ -661,37 +661,42 @@ shaka.text.UITextDisplayer = class {
     //   (for vertical growing left) is aligned at the line.
     // TODO: Implement line alignment with line number.
     // TODO: Implement lineAlignment of 'CENTER'.
-    if (cue.line != null) {
+    let line = cue.line;
+    if (line != null) {
       let lineInterpretation = cue.lineInterpretation;
       // HACK: the current implementation of UITextDisplayer only handled
-      // PERCENTAGE, as cue.line=0 is zero, we can interpret it as percentage
-      // safely.
-      if (cue.line == 0 &&
-          lineInterpretation == Cue.lineInterpretation.LINE_NUMBER) {
+      // PERCENTAGE, so we need convert LINE_NUMBER to PERCENTAGE
+      if (lineInterpretation == Cue.lineInterpretation.LINE_NUMBER) {
         lineInterpretation = Cue.lineInterpretation.PERCENTAGE;
+        const maxLines = 16;
+        if (line < 0) {
+          line = 100 + line / maxLines * 100;
```

Comment:
Negatively positioned cues will end up being positioned slightly differently than a positive positioned cue for the same location. For example, line:16 would be 100% but line:-1 would be at 93.75%. Probably not a huge different (assuming the 100% caption is still visible) but thought it worth pointing out.

Reply:
Do you think that for negative lines it is better: `line = (maxLines + line) / maxLines * 100;`?